right/src/macros/commands.c

```diff
@@ -1002,6 +1003,13 @@ static macro_result_t processIfSecondaryCommand(parser_context_t* ctx, bool nega
         strategy = SecondaryRoleStrategy_Advanced;
     }
 
+    if (ConsumeToken(ctx, "primaryFromSameHalf")) {
+        fromSameHalf = SecondaryRole_PrimaryFromSameHalf;
+    }
+    else if (ConsumeToken(ctx, "secondaryFromSameHalf")) {
+        fromSameHalf = SecondaryRole_SecondaryFromSameHalf;
+    }
```

Comment:
```suggestion
while(true) {
    if (ConsumeToken(ctx, "simpleStrategy")) {
        strategy = SecondaryRoleStrategy_Simple;
    }
    else if (ConsumeToken(ctx, "advancedStrategy")) {
        strategy = SecondaryRoleStrategy_Advanced;
    }
    else if (ConsumeToken(ctx, "primaryFromSameHalf")) {
        fromSameHalf = SecondaryRole_PrimaryFromSameHalf;
    }
    else if (ConsumeToken(ctx, "secondaryFromSameHalf")) {
        fromSameHalf = SecondaryRole_SecondaryFromSameHalf;
    }
    else {
        break;
    }
}
```

Let's allow any order?